### PR TITLE
[PlistHelper] Fall back to XML project output when Xcode initializati…

### PR DESCRIPTION
**Pull request by segiddins**
…on fails.

Closes https://github.com/CocoaPods/CocoaPods/issues/3723.

### CODE REVIEW

**Comment by neonichu** [comment]
Looks good in general, *but* it seems to only work with Xcode 7, because `IDEInitialize()` returns 0 on previous versions.

**Comment by neonichu** [comment]
BTW, this makes me think that it would be nice to have an env variable to completely disable loading of Xcode frameworks, so that users have an immediate workaround in case of breakage like this. I expected this to be a rather stable interface, but it evidently isn't.

**Comment by alloy** [comment]
Another option would be to have a list of supported Xcode versions. The pro is that the user doesn’t need to know of an option that’s likely to be hard to find, the con is that it requires a CP update. for new Xcode versions to work.

**Comment by alloy** [comment]
Or you could do both and have the env variable also be available to enable the functionality when running on an unsupported Xcode version.

**Comment by neonichu** [comment]
@alloy that sounds like a better idea

**Comment by alloy** [comment]
@neonichu lol, yah :D

**Comment by segiddins** [comment]
@neonichu want to take this over, then?

**Comment by neonichu** [comment]
@segiddins I would do the version detection as a separate PR - if you want, I can also finish this one.

**Comment by segiddins** [comment]
Yes please :)

**Comment by ddunbar** [comment]
If there is going to be an option for this, it might also be nice to have a way to not suppress stderr when calling IDEInitialize(), so it is more obvious what the exact failure is without having to modify the Xcodeproj code.

**Comment by segiddins** [comment]
@ddunbar maybe, but i'd honestly prefer just to print it out when `IDEInitialize` fails

**Comment by neonichu** [comment]
This should be good to merge now. It retains functionality for Xcode versions 7 beta 1 and earlier, while not breaking on 7 beta 2. However, it does not enable ASCII plist writing for 7 beta 2, opened #278 for tracking that.

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,10 @@
   [Samuel Giddins](https://github.com/segiddins)
   [#274](https://github.com/CocoaPods/Xcodeproj/issues/274)
 
+* Fall back to XML project output when Xcode initialization fails.  
+  [Samuel Giddins](https://github.com/segiddins)
```
Comment (segiddins) [inline CHANGELOG.md]: @neonichu add yourself :)

**Comment by segiddins** [comment]
:+1:

**Pull request merged**